package ProtoChampionship;

import "ProtoBattleDemo.proto";
import "DataCommon.proto";
import "ProtoBattle.proto";

//比赛玩家信息
message MatchUserInfoCPP
{
    required uint32 uid         = 1;
    required uint32 battleid    = 2;
    optional string fig         = 3; //头像
    optional string name        = 4; //名字
    optional uint32 grade       = 5; //段位
    optional uint32 heroid      = 6; //英雄id 
    optional uint32 status      = 7; //0:失败 1: 胜利
}

//比赛信息
message MatchInfoCPP
{
    repeated  MatchUserInfoCPP players = 1;  //所有比赛玩家
}


//战斗信息
//message BattleInfoCPP
//{
//    required  uint32    attack    = 1; //0: 防守, 1: 进攻
//    optional  ProtoBattleDemo.ProtoBattleDemo  battle       =   2; //战斗结果
//    optional  DataCommon.CommonItemsCPP    commons          =   3; //消耗
//}


message ProtoJoinInMatchReq
{
    required  uint32 id         = 1;  //场次id
    required  uint32 heroid     = 2;  //参加比赛的英雄
    required  uint32 free       = 3;  //0: 免费, 1: 付费
}

message ProtoJoinInMatchResp
{
    optional  DataCommon.CommonItemsCPP    commons                  =   1; //消耗
}

//开始比赛
message ProtoStartBattleReq
{
    repeated  uint32 props  = 1; //道具
}

//杯赛
message ProtoChampionshipReq
{
    required uint32 battleid                            = 1;; //
    optional ProtoBattle.BattleCommonCPP common         = 2;
}

message ProtoChampionshipResp
{
    required uint32 isAttacker                     = 1; //0:不是 1:是
    required ProtoBattle.ProtoBattleCPP common     = 2;
}

